addons/base/views.py

```diff
@@ -281,100 +283,208 @@ def get_auth(auth, **kwargs):
         sentry.log_message(str(err))
         raise HTTPError(http_status.HTTP_403_FORBIDDEN)
 
+
+def get_authenticated_resource(resource_id):
+    resource = AbstractNode.load(resource_id) or Preprint.load(resource_id)
+    if not resource:
+        raise HTTPError(http_status.HTTP_404_NOT_FOUND, message='Resource not found.')
+
+    # Convert a DraftNode to its corresponding node if applicable.
+    resource = resource.registered_draft.first() if isinstance(resource, DraftNode) else resource
+
+    if resource.deleted:
+        raise HTTPError(http_status.HTTP_410_GONE, message='Resource has been deleted.')
+
+    return resource
+
+
+def get_file_version_from_wb(waterbutler_data: dict) -> FileVersion:
+    path = waterbutler_data.get('path')
+    if not path:
+        return
```

Comment:
You are well and truly certain that this will kick out all non-osfstorage files?